ttnn/cpp/ttnn/operations/conv/conv2d/conv2d.cpp

```diff
@@ -671,9 +679,103 @@ std::tuple<ttnn::Tensor, uint32_t, uint32_t, ttnn::Tensor, std::optional<ttnn::T
+                    uint32_t output_slice_height = output_slice_height_end - output_slice_height_start;
+
+                    if(output_slice_height == 0) {
+                        continue;
+                    }
+
+                    int input_slice_height_start = output_slice_height_start * stride[0] - padding[0];
+                    int input_slice_height_end = (output_slice_height_end - 1) * stride[0] - padding[0] + (kernel_size[0] - 1) * (dilation[0] - 1) + kernel_size[0];
+                    int pad_top = std::max(0, -input_slice_height_start);
+                    int pad_bottom = std::max<int>(0, input_slice_height_end - input_height);
+                    input_slice_height_start = std::max(0, input_slice_height_start);
+                    input_slice_height_end = std::min<int>(input_height, input_slice_height_end);
+                    uint32_t input_slice_height = input_slice_height_end - input_slice_height_start;
+                    log_debug(tt::LogOp, "Output Slice : {}->{}", output_slice_height_start, output_slice_height_end);
+                    log_debug(tt::LogOp, "Input Slice : {}->{}", input_slice_height_start, input_slice_height_end);
+                    log_debug(tt::LogOp, "Padding : {}->{}", pad_top, pad_bottom);
+
+                    if(input_slice_height_start < input_slice_height_end) {
+                        auto sliced_input_tensor = ttnn::slice(input_tensor,
+                            std::array<uint32_t, 4>{batch_index, input_slice_height_start, 0, 0},  //Start
+                            std::array<uint32_t, 4>{batch_index, input_slice_height_end - 1, input_width - 1,in_channels - 1}, //End - Inclusive
+                            std::array<uint32_t, 4>{1, 1, 1, 1} //Step
+                            );
+                        log_debug(tt::LogOp, "Sliced input tensor shape: {}", sliced_input_tensor.get_shape());
+                        if(pad_top>0 || pad_bottom > 0)
+                        {
+                            auto pad_top_tensor = ttnn::pad(
+                                DefaultQueueId,
+                                sliced_input_tensor,
+                                std::vector<std::pair<uint32_t, uint32_t>>{{0, 0}, {pad_top, pad_bottom}, {0, 0}, {0, 0}},
+                                0, true, std::nullopt);
+                            sliced_input_tensor = pad_top_tensor;
+                        }
+                        log_debug(tt::LogOp, "Padded sliced input tensor shape: {}", sliced_input_tensor.get_shape());
+                        auto conv_config_l1 = conv_config;
+                        conv_config_l1.output_height_in_l1 = 0;
+                        conv_config_l1.reshard_if_not_optimal = true;
+                        ttnn::Tensor sliced_output_tensor;
+                        std::tie(sliced_output_tensor, std::ignore, std::ignore, weight_tensor_on_device, bias_tensor_on_device) = conv2d(
```

Comment:
1. This function is already too long. You should not add this much new code into it. Please refactor and have new code in a new function.
2. Do not make this function recursive. Please have multiple levels instead.